src/main/scala/mesosphere/marathon/storage/migration/Migration.scala

```diff
@@ -222,9 +222,9 @@ object Migration {
       },
       StorageVersions(1, 6, 0, StorageVersion.StorageFormat.PERSISTENCE_STORE) -> { (migration) =>
         new MigrationTo160(migration.instanceRepo, migration.persistenceStore)
-      }
-    // From here onwards we are not bound to the build version anymore.
-    //StorageVersions(200) -> { (migration) => new MigrationTo200(...) }
+      },
+      // From here onwards we are not bound to the build version anymore.
+      StorageVersions(200) -> { (migration) => new MigrationTo200(migration.instanceRepo, migration.persistenceStore) }
```

Comment:
Do we assign a range to each version then? Like `200` to 1.7.x, `300` to `1.8.x` and so forth? I am asking because it is unclear to me how we handle situations like the one for which we introduced `MigrationTo152`?

Reply:
The storage version is independent of the release version. However, we should either leave room to sneak in migrations or use the minor version for that.

So either

```
StorageVersion(200)
StorageVersion(201)
StorageVersion(250)
```

or
```
StorageVersion(200)
StorageVersion(200, 1)
StorageVersion(201)
```

Reply:
I like the first one better.

Reply:
Probably I am missing something. I don't get what benefit of fully decoupling storage versions from Marathon versions there is. What about having them like `1.7.x` where the first two numbers are coupled to the corresponding Marathon version, and `x` is just an incrementing number starting from `0`?

Reply:
The Storage version should be bound to a commit not a Marathon version. Especially with our new version scheme when we do not know the patch version the old versioning is hard.

Reply:
#5957 has more details.

Reply:
I still don't get the benefit of allocating storage version ranges to Marathon `major.minor` versions. I get why it is impossible to tie a storage `patch` version to a Marathon release one. But allocating `1.7.x` storage versions (where `x` is incremented manually), to `1.7.y` Marathon builds/versions (where `y` is the number of commits since a specific point in the commit history) has that benefit that it is clear right on in which `major.minor` version a particular storage migration is introduced. What do you think?

Reply:
I think I can understand both sides here. We're trying to decouple `major.minor.patch` from storage versions, but there's no strong reason why it needs to start with `2` or couldn't start with `1.7`.

Sure we can start with`StorageVersion(1, 7, 0)` and add `StorageVersion(1, 7, 1)` which might then be added in build artifact `1.7.120`. I can imagine there would be confusion wrt `StorageVersion(1, 7, 120)` and build artifact `1.7.120`, which are most likely not related at all.

From a storage migration perspective, the only thing that matters is that the values are incremental, unique, and ordered according to how they should be applied.

@ichernetsky from what I understand the only benefit from using `200` is to obviously disambiguate the version patch value from the migration patch value. A compromise could be to use `StorageVersion(17, 0)`, add additional migrations for 1.7 via `StorageVersion(17, 1)` etc, and move to `StorageVersion(18, 0)` for v1.8.

It's _probably_ less confusing than going to 200, and less confusing than having that storage version/built artifact mismatch.

@jeschkies WDYT?

Reply:
> @jeschkies WDYT?

In principle that's fine but what if we want to backport a migration? Suddenly we have a `StroageVersion(17)` on Marathon 1.6.